src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -96,13 +96,21 @@
   <Target Name="CopyNativeBinary" AfterTargets="Publish">
     <!-- replace apphost with binary we generated during native compilation -->
     <Delete Files="$(PublishDir)\$(TargetName)$(NativeBinaryExt)" />
-    <Copy SourceFiles="$(NativeOutputPath)$(TargetName)$(NativeBinaryExt)" DestinationFolder="$(PublishDir)" />
+    <Copy SourceFiles="$(NativeOutputPath)$(TargetName)$(NativeBinaryExt)" DestinationFolder="$(PublishDir)" Condition="'$(DotNetJsApi)' != 'true'" />
 
-    <Delete Files="$(PublishDir)\$(TargetName).js" Condition="'$(_targetOS)' == 'browser'"/>
-    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).js" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser'"/>
+    <Delete Files="$(PublishDir)\$(TargetName).js" Condition="'$(_targetOS)' == 'browser' and '$(DotNetJsApi)' != 'true'"/>
+    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).js" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser' and '$(DotNetJsApi)' != 'true'"/>
 
-    <Delete Files="$(PublishDir)\$(TargetName).wasm" Condition="'$(_targetOS)' == 'browser'"/>
-    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).wasm" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser'"/>
+    <Delete Files="$(PublishDir)\$(TargetName).wasm" Condition="'$(_targetOS)' == 'browser' and '$(DotNetJsApi)' != 'true'"/>
+    <Copy SourceFiles="$(NativeOutputPath)$(TargetName).wasm" DestinationFolder="$(PublishDir)" Condition="'$(_targetOS)' == 'browser' and '$(DotNetJsApi)' != 'true'"/>
+    
+    <ItemGroup>
+      <_FilesToPublish Include="$(IlcFrameworkPath)\dotnet*.js" />
+      <_FilesToPublish Include="$(IlcFrameworkPath)\dotnet*.map" />
+      <_FilesToPublish Include="$(_WasmIntermediateOutputPath)dotnet.native.js" />
+      <_FilesToPublish Include="$(_WasmIntermediateOutputPath)dotnet.native.wasm" />
```

Comment:
What does the output of a typical publish look like?

Reply:
Example output is here https://github.com/maraf/MinimalDotNetWasmNativeAOT/tree/dotnetjs/DotnetJsHack
- index.html & main.js is user code
- dotnet.js & dotnet.runtime.js are from runtime pack/ilc
  - .map are source maps pointing to typescript definitions on github
- dotnet.native.js & dotnet.native.wasm are emscripten outputs

So typically browser opens the index.html and that triggers main.js. User is then responsible for deciding if and how to start dotnet and what to invoke on it (it may or may not be main method)